### PR TITLE
Remove documentation captured on website

### PR DESCRIPTION
**Pull request by flands**
Addresses #2040

### CODE REVIEW

File: CONTRIBUTING.md
Comment (anuraaga) [inline CONTRIBUTING.md:129]: Thanks for the updates! The only thing that comes to mind is that snapshots are also often used by end users, not just contributors. It's probably ok though

**Review by anuraaga — APPROVED** [review]

File: CONTRIBUTING.md
Comment (jkwatson) [reply CONTRIBUTING.md:129]: Yeah, I'm not sure why we'd want to remove this information from the main README.

Comment (flands) [reply CONTRIBUTING.md:129]: Yeah, I went back and forth on this -- I would assume that releases are "stable" while snapshots may not be. I can add a link to this in the contributing file that way it is accessible.

Comment (flands) [reply CONTRIBUTING.md:129]: Addressed by: https://github.com/open-telemetry/opentelemetry.io/pull/310

**Comment by flands** [comment]
OK, website PR merged -- please review and merge this PR.

File: README.md
Comment (jkwatson) [inline README.md:13]: we no longer build on circleci. See the build link to GHAs above, which got removed.

Comment (jkwatson) [inline README.md:13]:
```suggestion
    <img alt="Build Status" src="https://github.com/open-telemetry/opentelemetry-java/workflows/Continuous%20Build/badge.svg">
```

File: README.md
Comment (jkwatson) [inline README.md:19]: this doesn't appear to be right, from the preview. Not sure exactly what it should be, though.

**Review by jkwatson — CHANGES_REQUESTED** [review]
Blocking this, pending decisions from the doc folks and the maintainers about how to keep documentation in sync.

**Comment by jkwatson** [comment]
In the maintainer meeting on Monday, I see this in the notes: "The bulk of our documentation should remain on GitHub." So, I'm going to close this PR pending a final decision on what should and should not be included in this repository vs. what is included on the website. We can re-open if we decide that the quickstart documentation should be removed.

**Pull request closed without merging**
